benches/bench.rs

```diff
@@ -1,44 +1,21 @@
 use criterion::{criterion_group, criterion_main, Bencher, Criterion};
 use interval_map::{Interval, IntervalMap};
+use rand::{rngs::StdRng, Rng, SeedableRng};
 use std::hint::black_box;
 
-struct Rng {
-    state: u32,
-}
-impl Rng {
-    fn new() -> Self {
-        Self { state: 0x87654321 }
-    }
-
-    fn gen_u32(&mut self) -> u32 {
-        self.state ^= self.state << 13;
-        self.state ^= self.state >> 17;
-        self.state ^= self.state << 5;
-        self.state
-    }
-
-    fn gen_range_i32(&mut self, low: i32, high: i32) -> i32 {
-        let d = (high - low) as u32;
-        low + (self.gen_u32() % d) as i32
-    }
-}
-
 struct IntervalGenerator {
-    rng: Rng,
-    limit: i32,
+    rng: StdRng,
 }
 impl IntervalGenerator {
     fn new() -> Self {
-        const LIMIT: i32 = 100000;
         Self {
-            rng: Rng::new(),
-            limit: LIMIT,
+            rng: StdRng::from_seed([0; 32]),
         }
     }
 
-    fn next(&mut self) -> Interval<i32> {
-        let low = self.rng.gen_range_i32(0, self.limit - 1);
-        let high = self.rng.gen_range_i32(low + 1, self.limit);
+    fn next(&mut self) -> Interval<u32> {
+        let low = self.rng.gen();
+        let high = self.rng.gen_range(low + 1..=u32::MAX);
         Interval::new(low, high)
     }
```

Comment:
Why do you remove the `limit`? In this case, the intervals in the benchmark will rarely overlapping.